bofire/surrogates/trainable.py

```diff
@@ -181,6 +182,22 @@ def cross_validate(
             # now do the scoring
             y_test_pred = self.predict(X_test)  # type: ignore
             y_train_pred = self.predict(X_train)  # type: ignore
+
+            # Convert to categorical if applicable
+            if isinstance(self.outputs.get_by_key(key).objective, ConstrainedCategoricalObjective):  # type: ignore
```

Comment:
Hmm, why only of `ConstrainedCategoricalObjective`? This should apply always for every `CategoricalOutput` feature, or?

Reply:
And in general, I am not 100% sure, what should be achieved here.

Reply:
CategoricalOutputs do not contain the functions to do the mapping (as per our previous discussions) from strings to categorical encodings. We need scalars for the loss function, not categories, for which Y would be an array of strings due to BoFire requiring that categorical columns be validated